### ISSUE TITLE
Should Alembic migration tests only be run if there's a diff in the alembic folder?

### ISSUE THREAD

**Issue by rdimaio** [rucio/rucio#7009]
For reference:

- https://github.com/rucio/rucio/tree/master/lib/rucio/db/sqla/migrate_repo/versions
- https://github.com/rucio/rucio/blob/8c4672c64b8784fd9f9cdc9e3516d7d83ecd22c9/tools/run_tests.sh#L32

These tests are always run at the beginning of every `run_tests.sh` call so far, which seems unnecessary.

**Comment by rdimaio** [comment]
Could use https://pypi.org/project/pytest-alembic/

**Comment by bari12** [comment]
Yes, doesn't need to be run every time. One time up/down to make sure the migrations work, but not for every call.

**Comment by maxnoe** [comment]
You also need to look at the db model definition. The upgrade/migrate tests should also detect a _missing_ migration script due to a schema change in the model definition.

**Comment by schallten** [comment]
Hi, I'd like to work on this. Is it still available?
